ui/components/TracePage/TracePage.tsx

```diff
@@ -867,45 +884,57 @@ const Trace = ({ traceId }: { traceId: string }) => {
     return () => window.removeEventListener("keydown", keyListener);
   }, [bindings]);
 
-  const [detailWidth, setDetailWidth] = useState(500);
-
   const firstRoot = getChildren(traceId)[0];
 
   const archerContainerRef = useRef<ArcherContainerHandle | null>(null);
   const refreshArcherArrows = useCallback(() => {
     archerContainerRef.current?.refreshScreen();
   }, []);
 
-  return (
-    <div className="flex flex-col h-full min-h-screen max-h-screen">
-      <div className="flex divide-x divide-gray-100 flex-1 overflow-clip">
-        <div className="flex-1 p-6 overflow-y-auto flex-shrink-0">
-          <Toolbar />
-          <ArcherContainer
-            ref={archerContainerRef}
-            noCurves
-            strokeColor="#E2E8F0"
-            strokeWidth={1}
-            startMarker={false}
-            endMarker={false}
-          >
-            {firstRoot ? (
-              <CallChildren id={firstRoot} refreshArcherArrows={refreshArcherArrows} />
-            ) : (
-              <div className="flex justify-center items-center h-full">
-                <Spinner size="medium" />
-              </div>
-            )}
-          </ArcherContainer>
-        </div>
+  const highlightedCalls = getHighlightedCalls(highlightedFunction, calls);
 
-        <Separator detailWidth={detailWidth} setDetailWidth={setDetailWidth} />
+  return (
+    <div className="h-screen">
+      <Allotment>
+        <Allotment vertical>
+          <div className="w-full h-full overflow-auto">
+            <div className="sticky top-0 z-10 bg-white p-4">
+              <Toolbar />
+            </div>
+            <div className="pl-4">
+              <ArcherContainer
+                ref={archerContainerRef}
+                noCurves
+                strokeColor="#E2E8F0"
+                strokeWidth={1}
+                startMarker={false}
+                endMarker={false}
+              >
+                {firstRoot ? (
+                  <CallChildren id={firstRoot} refreshArcherArrows={refreshArcherArrows} />
+                ) : (
+                  <div className="flex justify-center items-center h-full">
+                    <Spinner size="medium" />
+                  </div>
+                )}
+              </ArcherContainer>
+            </div>
+          </div>
+          <Allotment.Pane className="call-table" preferredSize={200}>
+            <Table
+              rows={highlightedCalls.map(({ fields = {} }) => fields)}
+              rowIds={highlightedCalls.map(({ id }) => id)}
+              onFocusChange={setSelectedId}
```

Comment:
I don't want `rowIndex` to leak out of the Table component because I'm getting rid of it in a follow-up due to [a bug](https://linear.app/ought/issue/ICE-45/fix-table-selection).